roles/version_update_single_node/tasks/main.yml

```diff
@@ -0,0 +1,90 @@
+---
+- name: Check if there is already an update in progress
+  scale_computing.hypercore.version_update_status_info:
+  register: update_status_before_update
+
+- name: Current update status
+  ansible.builtin.debug:
+    var: update_status_before_update
+
+- name: Get cluster info
+  scale_computing.hypercore.cluster_info:
+  register: cluster_info
+
+- name: Show cluster info
+  ansible.builtin.debug:
+    var: cluster_info
+
+- name: Get node info
+  scale_computing.hypercore.node_info:
+  register: node_info
+
+- name: Show node info
+  ansible.builtin.debug:
+    var: node_info
+
+- name: Check if single-node system - end host if False
+  ansible.builtin.meta: end_host
+  when: node_info.records | length > 1
+
+- name: Check if already on desired version - end host if True
+  ansible.builtin.meta: end_host
+  when: cluster_info.record.icos_version == scale_computing_hypercore_desired_version
+
+- name: Check if desired version available
+  scale_computing.hypercore.version_update_info:
+  register: available_updates
+
+- name: Show available updates
+  ansible.builtin.debug:
+    var: available_updates
+
+- name: End host if no update available
+  ansible.builtin.meta: end_host
+  when: available_updates.records == []
```

Comment:
Those 3 lines are redundunt ("Check if desired update is available - end host if Not available" does cover this case too), right? I will remove them. Ping back if you disagree.

Reply:
Agreed